### PR TITLE
Reintroduction of void installers

### PR DESCRIPTION
**Pull request by kinkinkijkin**
should work, test image before merge.

### CODE REVIEW

**Comment by kinkinkijkin** [comment]
tested and building. install test might need someone else.

**Comment by kinkinkijkin** [comment]
Can someone confirm working/unworking on devices other than krane?

File: fs/void-musl/build [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+	wget $ROOTFS_TAR -O void-$ARCH_UNAME.tar.xz # musl: void-aarch64, non-musl: void-aarch64-v
+
+fi
+tar xfp void-${ARCH_UNAME}.tar.xz -C $DESTDIR
+cp /etc/resolv.conf $DESTDIR/etc/resolv.conf
+chroot $DESTDIR $qemu /bin/xbps-install -Syu xbps
+chroot $DESTDIR $qemu /bin/xbps-install -yu
+chroot $DESTDIR $qemu /bin/xbps-install -y base-system
+chroot $DESTDIR $qemu /bin/xbps-remove -y base-voidstrap
+chroot $DESTDIR $qemu /bin/xbps-install -y vim dbus NetworkManager parted f2fs-tools eudev-libudev-devel base-devel git curl wget vboot-utils u-boot-tools
```
Comment (Maccraft123) [inline fs/void-musl/build:25]: drop this line

File: fs/void-musl/build [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+
+CADMIUMROOT=$(dirname $(dirname $(dirname $(realpath $0))))
+DESTDIR=$1
+
+. $CADMIUMROOT/config
+
+echo "Installing Void Linux (musl)"
+ROOTFS_TAR="$(curl -s -L https://voidlinux.org/download/ | grep $ARCH_UNAME | tr '"' ' ' | awk '{print $3}' | grep musl)"
+if [ ! -f "$ROOTFS_TAR" ]; then
+	rm void-${ARCH_UNAME}.tar.xz || true
```
Comment (Maccraft123) [inline fs/void-musl/build:15]: rename to `void-musl-${ARCH_UNAME}.tar.xz`

File: fs/void-musl/build [hunk 3]
```diff
@@ -0,0 +1,29 @@
+#!/bin/bash
+set -e
+set -x
+
+# !!!! TEMPORARY PLACEMENT, USE AT YOUR RISK !!!!
```
Comment (Maccraft123) [inline fs/void-musl/build:5]: drop this too

File: fs/void-musl/build [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+
+fi
+tar xfp void-${ARCH_UNAME}.tar.xz -C $DESTDIR
+cp /etc/resolv.conf $DESTDIR/etc/resolv.conf
+chroot $DESTDIR $qemu /bin/xbps-install -Syu xbps
+chroot $DESTDIR $qemu /bin/xbps-install -yu
+chroot $DESTDIR $qemu /bin/xbps-install -y base-system
+chroot $DESTDIR $qemu /bin/xbps-remove -y base-voidstrap
+chroot $DESTDIR $qemu /bin/xbps-install -y vim dbus NetworkManager parted f2fs-tools eudev-libudev-devel base-devel git curl wget vboot-utils u-boot-tools
+cp void-${ARCH_UNAME}.tar.xz $DESTDIR/void.tar.xz
```
Comment (Maccraft123) [inline fs/void-musl/build:26]: drop this line too, it's not needed

File: fs/void-musl/info [hunk 5]
```diff
@@ -0,0 +1,17 @@
+#!/bin/false
+
+# basic info
+FS_FANCYNAME="Void-musl"
```
Comment (Maccraft123) [inline fs/void-musl/info:4]: `Void with Musl libc`

File: fs/void-musl/info [hunk 6]
```diff
@@ -0,0 +1,17 @@
...
+#FS_HAS_OOBE="true"
```
Comment (Maccraft123) [inline fs/void-musl/info:5]: uncomment and set to `false`

File: fs/void-musl/info [hunk 7]
```diff
@@ -0,0 +1,17 @@
...
+FS_INST_PKG="/bin/xbps-install -y"
+FS_UPDATE="/bin/xbps-install -Syu"
+
+# bootloader packages
+FS_PKGS_CD_BOOTFW_UBOOT="u-boot-tools curl wget"
+FS_PKGS_CD_BOOTFW_DEPTHCHARGE="$FS_PKGS_CD_BOOTFW_UBOOT vboot-utils"
+
+# ui packages
```
Comment (Maccraft123) [inline fs/void-musl/info:13]: get those working if you can

File: fs/void-musl/info [hunk 8]
```diff
@@ -0,0 +1,17 @@
+#!/bin/false
+
+# basic info
+FS_FANCYNAME="Void-musl"
+#FS_HAS_OOBE="true"
+FS_INST_PKG="/bin/xbps-install -y"
+FS_UPDATE="/bin/xbps-install -Syu"
+
+# bootloader packages
+FS_PKGS_CD_BOOTFW_UBOOT="u-boot-tools curl wget"
```
Comment (Maccraft123) [inline fs/void-musl/info:10]: uhh, maybe we should make `FS_PKGS_CD_BASE`

**Review by Maccraft123 — CHANGES_REQUESTED** [review]
apply void-musl comments to void too

File: fs/void-musl/info (hunk 7 above)
Comment (kinkinkijkin) [reply fs/void-musl/info:13]: graphical environments require extra configuration on void that isn't done in advance on any other void installer, and would require a lot of void-specific code in install script.

File: fs/void-musl/info (hunk 8 above)
Comment (kinkinkijkin) [reply fs/void-musl/info]: needs to be added to debian as well

**Comment by kinkinkijkin** [comment]
Void installers updated, build and install updated.

File: fs/install [hunk 9]
```diff
@@ -180,21 +180,21 @@ if [ "$FS_HAS_OOBE" = "false" ]; then
...
-				chroot $DESTDIR $FS_INST_PKG $FS_PKGS_CD_UI_SWAY
+				chroot $DESTDIR $FS_INST_PKG $FS_PKGS_CD_UI_BASE $FS_PKGS_CD_UI_SWAY
 				;;
 			gnome)
-				chroot $DESTDIR $FS_INST_PKG $FS_PKGS_CD_UI_PHOSH
+				chroot $DESTDIR $FS_INST_PKG $FS_PKGS_CD_UI_BASE $FS_PKGS_CD_UI_GNOME
 				;;
 			kde)
-				chroot $DESTDIR $FS_INST_PKG $FS_PKGS_CD_UI_KDE
+				chroot $DESTDIR $FS_INST_PKG $FS_PKGS_CD_UI_BASE $FS_PKGS_CD_UI_KDE
```
Comment (Maccraft123) [inline fs/install:197]: unneded

File: fs/install [hunk 10]
```diff
@@ -180,21 +180,21 @@ if [ "$FS_HAS_OOBE" = "false" ]; then
...
 				chroot $DESTDIR passwd $USERNAME
 				;;
 			sway)
 				# TODO: config
-				chroot $DESTDIR $FS_INST_PKG $FS_PKGS_CD_UI_SWAY
+				chroot $DESTDIR $FS_INST_PKG $FS_PKGS_CD_UI_BASE $FS_PKGS_CD_UI_SWAY
 				;;
 			gnome)
-				chroot $DESTDIR $FS_INST_PKG $FS_PKGS_CD_UI_PHOSH
+				chroot $DESTDIR $FS_INST_PKG $FS_PKGS_CD_UI_BASE $FS_PKGS_CD_UI_GNOME
```
Comment (Maccraft123) [inline fs/install:194]: belongs in separate PR

File: fs/build [hunk 11]
```diff
@@ -36,7 +36,7 @@ fi
 
 $CADMIUMROOT/fs/$ROOTFS/build $DESTDIR
 source $CADMIUMROOT/fs/$ROOTFS/info
-$FS_INST_PKG $FS_PKGS_CD_BOOTFW_UBOOT $FS_PKGS_CD_BOOTFW_DEPTHCHARGE
+$FS_INST_PKG $FS_PKGS_CD_BASE $FS_PKGS_CD_BOOTFW_UBOOT $FS_PKGS_CD_BOOTFW_DEPTHCHARGE
```
Comment (Maccraft123) [inline fs/build:39]: belongs in separate PR

File: fs/void-musl/info [hunk 12]
```diff
@@ -0,0 +1,21 @@
...
+FS_PKGS_CD_BASE="vim dbus NetworkManager parted f2fs-tools eudev-libudev-devel base-devel git curl wget"
+# bootloader
+FS_PKGS_CD_BOOTFW_UBOOT="u-boot-tools"
+FS_PKGS_CD_BOOTFW_DEPTHCHARGE="$FS_PKGS_CD_BOOTFW_UBOOT vboot-utils"
+
+# ui
+FS_PKGS_CD_UI_BASE="xorg-fonts elogind dbus-elogind"
+FS_PKGS_CD_UI_GNOME="gnome gdm xorg-server-xwayland"
+#FS_PKGS_CD_UI_KDE="plasma-desktop kde-full krusader konsole plasma-nm muon kwin-x11 kwin-wayland sddm kinfocenter systemsettings powerdevil kde-config-sddm kde-config-screenlocker kde-config-gtk-style partitionmanager libkf5su-bin"
+#FS_PKGS_CD_UI_PHOSH="xfce4-terminal phosh phosh-tablet xwayland"
```
Comment (Maccraft123) [inline fs/void-musl/info:20]: are kde and phosh unsupported on void?

**Comment by kinkinkijkin** [comment]
Requesting re-review.

File: fs/install [hunk 13]
```diff
@@ -170,6 +170,7 @@ if [ "$FS_HAS_OOBE" = "false" ]; then
 
 	export USERNAME=$REPLY
 
+
```
Comment (Maccraft123) [inline fs/install:173]: drop this line

**Review by Maccraft123 — APPROVED** [review]
looks good, aside that one line

File: fs/install (hunk 13 above)
Comment (kinkinkijkin) [reply fs/install:173]: OOPS
